File: Selenium/Project/Activity6.py
import time

import pytest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


@pytest.fixture(scope="session")
def setup(request):


    driver = webdriver.Firefox()


    session = request.node


    for item in session.items:
        cls = item.getparent(pytest.Class)
        setattr(cls.obj,"driver",driver)

    # Execute the test method
    yield

    # Close the browser once the test has ended
    request.addfinalizer(driver.close)

@pytest.mark.usefixtures("setup")
class TestSite:

    def test_home_page(self):
        self.driver.get("https://alchemy.hguy.co/jobs/")
        self.driver.maximize_window()
        #click on job
        JobsPage=self.driver.find_element(By.LINK_TEXT,"Jobs")
        JobsPage.click()
        time.sleep(2)
        #providing Keyword in Textbox
        searchkey=self.driver.find_element(By.XPATH,"//input[@id='search_keywords']")
        searchkey.send_keys("Banking")
        time.sleep(2)
        # click on search button
        searchbutton=self.driver.find_element(By.XPATH,"//input[@type='submit']")
        searchbutton.click()
        time.sleep(2)
        #selecting the BankingTestJob result
        result=self.driver.find_element(By.XPATH,"//div[@class='position']/h3[text()='BankingTestJob']")
        result.click()
        time.sleep(2)
        #click on apply job
        Apply=self.driver.find_element(By.XPATH,"//input[@type='button']")
        Apply.click()
        time.sleep(2)
        #prinitng the email id
        Email=self.driver.find_element(By.XPATH,"//a[text()='abhiram@cklabs.com']")
        print("\nEmail id is :" +Email.text)

File: Selenium/Project/Activity6.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.*;

import java.util.List;
import java.util.Set;


public class Activity6 {


    WebDriver driver = new FirefoxDriver();

        @BeforeMethod
        public void openBrowser() throws InterruptedException {
            WebDriverManager.firefoxdriver().setup();
            driver.get("https://alchemy.hguy.co/jobs/");
            driver.manage().window().maximize();
            Thread.sleep(1000);
        }

        @Test
        public void JobPages() throws InterruptedException {
            WebElement JobsPage= driver.findElement(By.linkText("Jobs"));
            JobsPage.click();
            Thread.sleep(1000);

        }

       // @Test(dependsOnMethods = {"JobPages"})
       @AfterMethod
        public void SearchJob() throws InterruptedException {
            //sending input to Keyword inputtext
            WebElement searchkey=driver.findElement(By.xpath("//input[@id='search_keywords']"));
            searchkey.sendKeys("Banking");
           Thread.sleep(1000);

            //clicking on search job button
            WebElement searchbutton=driver.findElement(By.xpath("//input[@type='submit']"));
            searchbutton.click();
           Thread.sleep(1000);
            //selecting the BankingTestJob result
            WebElement result=driver.findElement(By.xpath("//div[@class='position']/h3[text()='BankingTestJob']"));
            result.click();
           Thread.sleep(1000);


        }
        @AfterClass
        public void ApplyJob() throws InterruptedException {
            WebElement Apply=driver.findElement(By.xpath("//input[@type='button']"));
            Apply.click();
            Thread.sleep(1000);
            WebElement Email=driver.findElement(By.xpath("//a[text()='abhiram@cklabs.com']"));
            System.out.println("Email id is :" +Email.getText());
        }

      @AfterTest
        public void closeBrowser(){
            driver.quit();

        }

}

